packages/components/src/components/hds/copy/snippet/index.ts

```diff
@@ -7,17 +7,47 @@ import Component from '@glimmer/component';
 import { assert } from '@ember/debug';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
+import { HdsCopySnippetColorValues } from './types.ts';
+import type { HdsCopySnippetColors } from './types.ts';
+import type { HdsClipboardModifierSignature } from '../../../../modifiers/hds-clipboard.ts';
+
+export const DEFAULT_COLOR = HdsCopySnippetColorValues.Primary;
+export const COLORS: string[] = Object.values(HdsCopySnippetColorValues);
 
-export const DEFAULT_COLOR = 'primary';
-export const COLORS = ['primary', 'secondary'];
 export const DEFAULT_ICON = 'clipboard-copy';
 export const SUCCESS_ICON = 'clipboard-checked';
 export const ERROR_ICON = 'clipboard-x';
 export const DEFAULT_STATUS = 'idle';
 
-export default class HdsCopySnippetIndexComponent extends Component {
+interface HdsCopySnippetSignature {
+  Args: {
+    color?: HdsCopySnippetColors;
+    isFullWidth?: boolean;
+    isTruncated?: boolean;
+    onError?: HdsClipboardModifierSignature['Args']['Named']['onError'];
+    onSuccess?: HdsClipboardModifierSignature['Args']['Named']['onSuccess'];
+    textToCopy: HdsClipboardModifierSignature['Args']['Named']['text'];
+  };
+  Element: HTMLButtonElement;
+}
+
+export default class HdsCopySnippetComponent extends Component<HdsCopySnippetSignature> {
   @tracked status = DEFAULT_STATUS;
-  @tracked timer;
+  @tracked timer: ReturnType<typeof setTimeout> | undefined;
+
+  /**
+   * @param textToCopy
+   * @type {string | number | bigint | undefined} ???
+   */
+  get textToCopy() {
```

Comment:
I want to discuss this with @alex-ju (I see you've done to overcome TS complaining about `bigint` not being assignable to type 'ContentValue'). I would be keen to drop support for `bigint` in the `hds-modifier` to avoid this extra layer of complexity (consumers can always easily convert a `bigint` to a `string`.

Reply:
OK, I understand why the confusion (my confusion, I mean). the `@textToCopy` (argument) is passed down to the `hds-clipboard` modifier, while `this.textToCopy` is the one used to show the value at UI level. My suggestion is to rename the getter to avoid this confusion, something like this:
```suggestion
  get textToShow() {
```